src/main/java/bio/terra/cda/app/controller/MetaApiController.java

```diff
@@ -30,16 +36,27 @@ public MetaApiController(ApplicationConfiguration applicationConfiguration) {
 
   @Override
   public ResponseEntity<SystemStatus> serviceStatus() {
-    HttpStatus httpStatus = HttpStatus.OK;
-
-    SystemStatusSystems otherSystemStatus =
-        new SystemStatusSystems().ok(true).addMessagesItem("everything is fine");
+    SystemStatus systemStatus = new SystemStatus();
+    try {
+      BigQuery bigQuery = BigQueryOptions.newBuilder().setProjectId("gdc-bq-sample").build().getService();
+      String StatusCheck = bigQuery.getDataset("cda_mvp").getDatasetId().getDataset();
+      if (StatusCheck.equals("cda_mvp")) {
+        SystemStatusSystems otherSystemStatus = new SystemStatusSystems().ok(true).addMessagesItem("everything is fine");
+        systemStatus.ok(true).putSystemsItem("BigQuery", otherSystemStatus);
+        return new ResponseEntity<>(systemStatus, HttpStatus.OK);
+      }
+    } catch (Exception e) {
+      logger.error(e.getMessage());
```

Comment:
When logging exceptions, you almost always want to log the exception itself, so the log will show the stack trace from the exception.
```suggestion
      logger.error("status check failed", e);
```